### PR TITLE
Functionality to show a random quote

### PR DESCRIPTION
**Pull request by salindholm**
[API provides functionality to show a random quote](https://www.pivotaltracker.com/n/projects/2478349)
### User Story
```
As a developer
In order to be able to show a random quote on our website
We would like to be able to get a random quote from an external API
```
## Changes proposed in this pull request:
* Adding rest-client and generating a Quotes controller.
* Creating a test to check if API gets a random quote from our external Api
* Adding web-mock and configuring it in rails_helper, and adding Show action to Quotes controller.
* Creating a fixture file for random quote.
* Stubbing out fixture path and disabling WebMock net connect and to allow localhost in rails_helper.rb

## What I have learned working on this feature:
* How to work with web-mock and rest-client
* How to work with an external API

## Packages/Gems added
* Web-mock
* Rest-client

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,6 +7,8 @@ gem 'rails', '~> 6.0.3', '>= 6.0.3.4'
 gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 4.1'
 gem 'bootsnap', '>= 1.4.2', require: false
+gem 'rest-client'
+gem 'webmock'
```
Comment (tochman) [inline Gemfile]: This gem is NOT going to be used in production. Please move to `:development, :test`

File: Gemfile [hunk 2]
```diff
@@ -7,6 +7,8 @@ gem 'rails', '~> 6.0.3', '>= 6.0.3.4'
...
 
 group :development, :test do
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
```
Comment (tochman) [inline Gemfile]: Are you using `byebug` ?

File: app/controllers/quotes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class QuotesController < ApplicationController
+  API_URL = 'https://api.tronalddump.io'.freeze
+
+  def show
+    if params[:id] == 'random'
+      url = "#{API_URL}/random/quote"
+      result = RestClient.get(url)
+
```
Comment (tochman) [inline app/controllers/quotes_controller.rb]: Please remove empy lines

File: app/controllers/quotes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    render json: format_json_response(quote)
+  end
+
+  private
+
+  def format_json_response(obj)
+    {
+      date: DateTime.parse(obj['appeared_at']).strftime('%B %Y'),
+      source: obj['_embedded']['source'][0]['url'],
+      quote_of_the_day: obj['value']
```
Comment (tochman) [inline app/controllers/quotes_controller.rb]: Is the key name `quote_of_the_day` appropriate in your case?

File: config/routes.rb [hunk 5]
```diff
@@ -1,2 +1,3 @@
 Rails.application.routes.draw do
+  resources :quotes, only: [:show]
```
Comment (tochman) [inline config/routes.rb]: I see that you chose to NOT use any namespaces on this API application. Can you give me a reason for that? Not a blocker, I just want to know.

File: spec/requests/get_five_random_trump_quotes_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  describe 'with "random" as query parameters' do
+    before do
+      get '/quotes/random'
+    end
+
+    it 'is expected to return a quote' do
+      expected_response_body = {
+        "date": 'October 2016',
+        "source": 'https://twitter.com/realDonaldTrump/status/787699930718695425',
+        "quote_of_the_day": 'The election is absolutely being rigged by the dishonest and distorted media pushing Crooked Hillary - but also at many polling places - SAD'
```
Comment (tochman) [inline spec/requests/get_five_random_trump_quotes_spec.rb]: Again, I wonder about the naming here...

File: spec/requests/get_five_random_trump_quotes_spec.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+      }.as_json
+
+      expect(JSON.parse(response.body)).to eq expected_response_body
```
Comment (tochman) [inline spec/requests/get_five_random_trump_quotes_spec.rb]: Don't you have a helper method to parse the body into `json`? If not, I would suggest that you add one. Take a look at the Cooper challenge documentation.

**Review by tochman — CHANGES_REQUESTED** [review]
A would like you to do a few stylistic changes...

File: spec/requests/get_five_random_trump_quotes_spec.rb (hunk 7 above)
Comment (salindholm) [reply spec/requests/get_five_random_trump_quotes_spec.rb]: That is true, we have now added the ResponseJSON helper and changed the test accordingly!

File: spec/requests/get_five_random_trump_quotes_spec.rb (hunk 6 above)
Comment (salindholm) [reply spec/requests/get_five_random_trump_quotes_spec.rb]: This is fixed as mentioned!

File: config/routes.rb (hunk 5 above)
Comment (salindholm) [reply config/routes.rb]: We decided to go for api but we forgot to update it, our bad... The front end is already using the api route

File: app/controllers/quotes_controller.rb (hunk 4 above)
Comment (salindholm) [reply app/controllers/quotes_controller.rb]: Not really.. we have changed it to random_quote, thanks for the input!

File: app/controllers/quotes_controller.rb (hunk 3 above)
Comment (salindholm) [reply app/controllers/quotes_controller.rb]: Noted, its now deleted

File: Gemfile (hunk 2 above)
Comment (salindholm) [reply Gemfile]: No, so we have removed it. We will use Pry for debugging

File: Gemfile (hunk 1 above)
Comment (salindholm) [reply Gemfile]: This is now moved accordingly!

File: Gemfile [hunk 8]
```diff
@@ -9,12 +9,13 @@ gem 'puma', '~> 4.1'
 gem 'bootsnap', '>= 1.4.2', require: false
 
 group :development, :test do
-  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
   gem 'rspec-rails'
   gem 'shoulda-matchers'
   gem 'factory_bot_rails'
   gem 'pry-rails'
   gem 'coveralls', require: false
+  gem 'rest-client'
```
Comment (tochman) [inline Gemfile]: Haha.. Noooo. `rest-client` is used in production. `webmock` is not.

Comment (salindholm) [reply Gemfile]: Oh, we thought it was regarding both gems added, sorry haha! Now rest-client is back where it belongs!

**Pull request merged**